### PR TITLE
dcos-2523 respect DCOS_LOG_LEVEL and DCOS_DEBUG env variables

### PR DESCRIPTION
**Pull request by tamarrow**

### CODE REVIEW

File: cli/dcoscli/main.py [hunk 1]
```diff
@@ -89,43 +91,57 @@ def _main():
...
     :rtype: bool
     """
 
-    if log_level is None:
+    if unset:
         os.environ.pop(constants.DCOS_LOG_LEVEL_ENV, None)
         return True
 
-    log_level = log_level.lower()
+    log_level = ((log_level and log_level.lower()) or
```
Comment (mgummelt) [inline cli/dcoscli/main.py]: if you're trying to do a ternary operation, use `a if b else c`

File: cli/dcoscli/main.py [hunk 2]
```diff
@@ -89,43 +91,57 @@ def _main():
...
+                 os.environ.get(constants.DCOS_LOG_LEVEL_ENV, None))
```
Comment (mgummelt) [inline cli/dcoscli/main.py]: The `None` argument is redundant.  `get` returns None by default if it can't find the key

File: cli/dcoscli/main.py [hunk 3]
```diff
@@ -89,43 +91,57 @@ def _main():
...
     if log_level in constants.VALID_LOG_LEVEL_VALUES:
         os.environ[constants.DCOS_LOG_LEVEL_ENV] = log_level
         return True
 
+    if log_level is None:
```
Comment (mgummelt) [inline cli/dcoscli/main.py]: place this before the previous conditional

File: cli/dcoscli/main.py [hunk 4]
```diff
@@ -89,43 +91,57 @@ def _main():
...
+        return True
+
     msg = 'Log level set to an unknown value {!r}. Valid values are {!r}'
```
Comment (mgummelt) [inline cli/dcoscli/main.py]: This should raise an Exception

File: cli/dcoscli/main.py [hunk 5]
```diff
@@ -89,43 +91,57 @@ def _main():
     subproc = Popen([executable,  command] + args['<args>'],
                     stderr=PIPE)
 
+    if log_level:
```
Comment (mgummelt) [inline cli/dcoscli/main.py]: why unset?

**Comment by mgummelt** [comment]
tests?

File: cli/dcoscli/main.py (hunk 1 above)
Comment (tamarrow) [inline cli/dcoscli/main.py]: done

File: cli/dcoscli/main.py (hunk 2 above)
Comment (tamarrow) [inline cli/dcoscli/main.py]: done

File: cli/dcoscli/main.py (hunk 3 above)
Comment (tamarrow) [inline cli/dcoscli/main.py]: done

File: cli/dcoscli/main.py (hunk 5 above)
Comment (tamarrow) [inline cli/dcoscli/main.py]: if they specify --log-level we unset the env variable so it doesn't persist logging. We only persist if they explicitly set the env variable

File: cli/dcoscli/main.py (hunk 4 above)
Comment (tamarrow) [inline cli/dcoscli/main.py]: done

File: cli/dcoscli/main.py (hunk 5 above)
Comment (mgummelt) [inline cli/dcoscli/main.py]: what do you mean "persist logging"?

Comment (tamarrow) [inline cli/dcoscli/main.py]: persist ie keep logging with that level for all commands until they unset the env variable

Comment (mgummelt) [inline cli/dcoscli/main.py]: env vars are only set for the current process.  they don't persist in the parent process.

Comment (tamarrow) [inline cli/dcoscli/main.py]: yes, but if they set it in that process shouldn't we keep logging on for that process?

**Comment by tamarrow** [comment]
added tests

**Comment by tamarrow** [comment]
@mgummelt simplified pr

File: cli/dcoscli/main.py [hunk 6]
```diff
@@ -101,10 +101,10 @@ def _config_log_level_environ(log_level):
     """
 
     if log_level is None:
```
Comment (mgummelt) [inline cli/dcoscli/main.py]: If this function does nothing on None, you should probably just have the client code check for None, and then not call

File: cli/dcoscli/main.py [hunk 7]
```diff
@@ -124,8 +124,8 @@ def _config_debug_environ(is_debug):
 
     if is_debug:
         os.environ[constants.DCOS_DEBUG_ENV] = 'true'
-    else:
-        os.environ.pop(constants.DCOS_DEBUG_ENV, None)
+
+    return None
```
Comment (mgummelt) [inline cli/dcoscli/main.py]: we usually don't explicitly return None

File: cli/dcoscli/main.py [hunk 8]
```diff
@@ -124,8 +124,8 @@ def _config_debug_environ(is_debug):
 
     if is_debug:
```
Comment (mgummelt) [inline cli/dcoscli/main.py]: this function is so simple, it should probably just be removed, but up to you

**Comment by mgummelt** [comment]
minor comments then LGTM

File: cli/dcoscli/main.py (hunk 6 above)
Comment (tamarrow) [inline cli/dcoscli/main.py]: done

File: cli/dcoscli/main.py (hunk 7 above)
Comment (tamarrow) [inline cli/dcoscli/main.py]: removed

File: cli/dcoscli/main.py (hunk 8 above)
Comment (tamarrow) [inline cli/dcoscli/main.py]: true, removed

**Comment by tamarrow** [comment]
thanks @mgummelt. made changes

**Pull request merged**